backends/ref/ceed-ref-restriction.c

```diff
@@ -738,12 +781,16 @@ int CeedElemRestrictionCreate_Ref(CeedMemType mem_type, CeedCopyMode copy_mode,
   CeedCallBackend(CeedElemRestrictionSetData(rstr, impl));
   CeedCallBackend(CeedElemRestrictionSetELayout(rstr, layout));
   CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "Apply", CeedElemRestrictionApply_Ref));
-  CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "ApplyUnsigned", CeedElemRestrictionApplyUnsigned_Ref));
-  CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "ApplyUnoriented", CeedElemRestrictionApplyUnoriented_Ref));
-  CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "ApplyBlock", CeedElemRestrictionApplyBlock_Ref));
+  if (rstr_type == CEED_RESTRICTION_POINTS) {
+    CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "ApplyAtPointsInElement", CeedElemRestrictionApplyAtPoints_Ref));
+  } else {
+    CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "ApplyUnsigned", CeedElemRestrictionApplyUnsigned_Ref));
+    CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "ApplyUnoriented", CeedElemRestrictionApplyUnoriented_Ref));
+    CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "ApplyBlock", CeedElemRestrictionApplyBlock_Ref));
+    CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "GetOrientations", CeedElemRestrictionGetOrientations_Ref));
+    CeedCallBackend(CeedSetBackendFunction(ceed, "ElemRestriction", rstr, "GetCurlOrientations", CeedElemRestrictionGetCurlOrientations_Ref));
+  }
```

Comment:
I'm not sure I get this, as other "special" element restrictions (like the specially oriented ones) don't get different treatment.

Reply:
I guess it's this way because the restriction should only be used via `CeedElemRestrictionApplyAtPointsInElement` and not by anything else?

Reply:
Yeah, the use of the indices is fundamentally different in the case of the `AtPoints` restriction, while for all the other restrictions we are including or excluding orientation information but everything else means the same thing. I think it would be harmless to drop the `else` there.

Reply:
Gotcha. I'm good either way.

Reply:
I'll leave this unresolved as a reminder in case you did want to remove the `else`. Otherwise, feel free to resolve and merge.

Reply:
I went ahead and dropped the `else`